app/models/manageiq/providers/vmware/cloud_manager/refresh_parser.rb

```diff
@@ -0,0 +1,123 @@
+class ManageIQ::Providers::Vmware::CloudManager::RefreshParser < ManageIQ::Providers::CloudManager::RefreshParser
+  def self.ems_inv_to_hashes(ems, options = nil)
+    new(ems, options).ems_inv_to_hashes
+  end
+
+  def initialize(ems, options = Config::Options.new)
+    @ems                 = ems
+    @data                = {}
+    @data_index          = {}
+    @options             = options
+
+    @connection          = ems.connect
+
+    @inv = Hash.new { |h, k| h[k] = [] }
+  end
+
+  def process_collection(collection, key)
+    @data[key] ||= []
+
+    collection.each do |item|
+      uid, new_result = yield(item)
+      next if uid.nil?
+
+      @data[key] << new_result
+      @data_index.store_path(key, uid, new_result)
+    end
+  end
+
+  def ems_inv_to_hashes
+    log_header = "MIQ(#{self.class.name}.#{__method__}) Collecting data for EMS name: [#{@ems.name}] id: [#{@ems.id}]"
+
+    $log.info("#{log_header}...")
+
+    @ems.api_version = @connection.api_version
+
+    @inv[:orgs] = @connection.organizations.all.to_a
+
+    @inv[:orgs].each do |org|
+      @inv[:vdcs] += org.vdcs.all
+    end
+
+    @inv[:vdcs].each do |vdc|
+      @inv[:vapps] += vdc.vapps.all
+    end
+
+    @inv[:vapps].each do |vapp|
+      @inv[:vms] += vapp.vms.all
+    end
```

Comment:
Its okay that they rely on each other since they all use `@inv[:key]`, but if you don't want to use get_xyz then I'd rather pull that out into a `retrieve_inventory` or something like that rather than sticking it all in `get_vms`